[Docs] E2E Preparation for Beta TestNet release (#932)
pokt-network/poktroll: docusaurus/docs/explore/rpc.md

sidebar_position: 3
---

- [Types of RPC Endpoints](#types-of-rpc-endpoints)
- [Beta TestNet](#beta-testnet)
  - [Beta RPC Endpoints](#beta-rpc-endpoints)
  - [Beta JSON-RPC Example](#beta-json-rpc-example)
- [Alpha TestNet](#alpha-testnet)
  - [Alpha RPC Endpoints](#alpha-rpc-endpoints)
  - [Alpha JSON-RPC Example](#alpha-json-rpc-example)

## Types of RPC Endpoints

> We have provided `gRPC`, `JSON-RPC` and `REST` endpoints, which are available here:

Reviewer: ```suggestion
```

You can review the difference between them in the [Cosmos SDK docs](https://docs.cosmos.network/main/learn/advanced/grpc_rest#comparison-table).

## Beta TestNet

### Beta RPC Endpoints

> We have provided `gRPC`, `JSON-RPC` and `REST` endpoints, which are available here:

Reviewer: ```suggestion
We provide `gRPC`, `JSON-RPC` and `REST` endpoints, which are available here:
```

poktrolld query block --type=height 0 --node https://shannon-testnet-grove-seed-rpc.poktroll.com
```

## Alpha TestNet

### Alpha RPC Endpoints

> We have provided `gRPC`, `JSON-RPC` and `REST` endpoints, which are available here:

Reviewer: ```suggestion
We provide `gRPC`, `JSON-RPC` and `REST` endpoints, which are available here:
```

> - **RPC**: `https://shannon-testnet-grove-seed-rpc.alpha.poktroll.com`

Reviewer 2: These are not active currently, but soon will be - I've already added the changes. Need to apply/test.

- **gRPC**: `https://shannon-testnet-grove-seed-grpc.alpha.poktroll.com`
- **REST**: `https://shannon-testnet-grove-seed-api.alpha.poktroll.com`

### Alpha JSON-RPC Example

Using `curl`:

```bash

> curl -X POST https://shannon-testnet-grove-seed-rpc.poktroll.com/block

Reviewer 2: ```suggestion
curl -X POST https://shannon-testnet-grove-seed-rpc.alpha.poktroll.com/block
```

```

Using the `poktrolld` binary:

```bash

> poktrolld query block --type=height 0 --node https://shannon-testnet-grove-seed-rpc.poktroll.com

Reviewer: ```suggestion
poktrolld query block --type=height 1 --node https://shannon-testnet-grove-seed-rpc.poktroll.com
```

Height `0` returns an error

> poktrolld query block --type=height 0 --node https://shannon-testnet-grove-seed-rpc.poktroll.com

Reviewer 2: ```suggestion
poktrolld query block --type=height 0 --node https://shannon-testnet-grove-seed-rpc.alpha.poktroll.com
```

> poktrolld query block --type=height 0 --node https://shannon-testnet-grove-seed-rpc.alpha.poktroll.com

Reviewer: ```suggestion
poktrolld query block --type=height 1 --node https://shannon-testnet-grove-seed-rpc.alpha.poktroll.com
```
Using height `0` returns an error. Could you please update it for all occurrences in this doc?
